src/profiler/profiler.h

```diff
@@ -1149,8 +1158,15 @@ struct ProfileOperator : public ProfileEvent {
     : ProfileEvent(name)
       , as_task_(name, &domain_)
       , name_(name)
-      , attributes_(attributes) {
-    SetCategories(domain_.name());
+      , attributes_(attributes)
+      , profiling_operator_(strcmp(name, "Dummy_Wait") && strcmp(name, "Custom"))
+      , profiling_task_(strcmp(name, "Dummy_Wait") && strcmp(name, "Custom")) {
+    if (strstr(name, "::")) {
+      as_task_.setDomain(&custom_op_domain);
```

Comment:
Seems this line can be merged with https://github.com/apache/incubator-mxnet/pull/15210/files#diff-b2c0746deeb2e04ea9cdd8139d6b30b2R1159

Reply:
In this case you are calling as_task_ twice with the second call overwrites the previous one. Using a ternary operator during class initialization seems cleaner to me.

Reply:
I still think that because line 1165 and line 1166 are basically doing the same thing, except for different objects, we should not move 1165 up. Also, I think keeping the initialization list concise is also good

Reply:
I think a good design style is actually keep the body in constructor simple. In fact, if you do
```suggestion
      as_task_(name, IsSubOperatorOfCustom(name) ? &custom_op_domain :  &domain_
```
Then you don't even need a if/else branch for SetCategories. You can simply write:
```suggestion
      SetCategories(domain_.name())
```
because the domain_ here is already customer_domain in your if case.

Reply:
Sure.

Reply:
domain_ is a global variable, not a member variable of ProfileOperator